internal/serverinstall/serverinstall.go

```diff
@@ -8,15 +8,67 @@ import (
 	"github.com/hashicorp/waypoint/internal/clicontext"
 	"github.com/hashicorp/waypoint/internal/pkg/flag"
 	pb "github.com/hashicorp/waypoint/internal/server/gen"
+	"github.com/hashicorp/waypoint/internal/serverconfig"
 )
 
 // Installer is implemented by the server platforms and is responsible for managing
 // the installation of the Waypoint server.
 type Installer interface {
-	Install(ctx context.Context, ui terminal.UI, log hclog.Logger) (*clicontext.Config, *pb.ServerConfig_AdvertiseAddr, string, error)
+	// Install expects the Waypoint server to be installed.
+	Install(context.Context, *InstallOpts) (*InstallResults, error)
+
+	// InstallRunner expects a Waypoint runner to be installed.
+	InstallRunner(context.Context, *InstallRunnerOpts) error
```

Comment:
I had to make this a separate function call because we need to have an auth token for the runner, and to get an auth token we have to be bootstrapped, and the bootstrapped happens in the shared `waypoint install` logic _after_ `Install`. 

So the flow is: (1) p.Install (2) common logic bootstraps and gets runner token (3) call p.InstallRunner.

Reply:
TOTALLY. I almost wonder if we should call it `InstallPhaseTwo` on the off hand there could be further components. But we can always change the name WHEN we have further components.

Reply:
Yeah this is a non-pluggable thing (just internally extendable) so I think we can cross that bridge later.